packages/cli/src/task-runners/task-broker/task-broker-ws-server.ts

```diff
@@ -99,7 +99,9 @@ export class TaskBrokerWsServer {
 
 		const onMessage = async (data: WebSocket.RawData) => {
 			try {
-				const buffer = Array.isArray(data) ? Buffer.concat(data) : Buffer.from(data);
+				const buffer = Array.isArray(data)
+					? Buffer.concat(data)
+					: Buffer.from(data instanceof ArrayBuffer ? new Uint8Array(data) : data);
```

Comment:
cubic has a point here. When we wrap `data` in an `UInt8Array` it causes an unnecessary memory copy (see [this](https://nodejs.org/docs/latest-v20.x/api/buffer.html#static-method-bufferfrombuffer)). I think it's because the `Buffer.from` overload fails to infer the correct usage. This works and doesn't do extra allocations (tho not the prettiest):

```ts
const buffer = Array.isArray(data)
	? Buffer.concat(data)
	: data instanceof ArrayBuffer
		? Buffer.from(data)
		: data;
```

It also seems that we have done the memory copy all this time without realizing (as `Buffer.from` copies when the input is another instanceof `Buffer`). We could fix this now or merge this first and I can fix it afterwards.